yarn-project/archiver/src/archiver/eth_log_handlers.ts

```diff
@@ -1,16 +1,44 @@
 import { Hex, Log, PublicClient, decodeFunctionData, getAbiItem, getAddress, hexToBytes } from 'viem';
-import { RollupAbi, UnverifiedDataEmitterAbi } from '@aztec/l1-artifacts';
+import { InboxAbi, RollupAbi, UnverifiedDataEmitterAbi } from '@aztec/l1-artifacts';
+import { Fr } from '@aztec/foundation/fields';
 import {
   BufferReader,
   ContractData,
   ContractPublicData,
   EncodedContractFunction,
+  L1ToL2Message,
+  L1Actor,
+  L2Actor,
   L2Block,
   UnverifiedData,
 } from '@aztec/types';
 import { EthAddress } from '@aztec/foundation/eth-address';
 import { AztecAddress } from '@aztec/foundation/aztec-address';
 
+/**
+ * Processes newly received MessageAdded (L1 to L2) logs.
+ * @param logs - MessageAdded logs.
+ * @returns Array of all Pending L1 to L2 messages that were processed
+ */
+export function processPendingL1ToL2MessageAddedLogs(
+  logs: Log<bigint, number, undefined, typeof InboxAbi, 'MessageAdded'>[],
+): L1ToL2Message[] {
+  const l1ToL2Messages: L1ToL2Message[] = [];
+  for (const log of logs) {
+    l1ToL2Messages.push(
```

Comment:
stylistic nit but maybe destructure the log at the top? then you wont have to repeat `log.args` everywhere.